pkg/host/internal/network/network.go

```diff
@@ -172,16 +178,42 @@ func (n *network) SetNetdevMTU(pciAddr string, mtu int) error {
 
 func (n *network) GetNetDevMac(ifaceName string) string {
 	log.Log.V(2).Info("GetNetDevMac(): get Mac", "device", ifaceName)
-	macFilePath := filepath.Join(vars.FilesystemRoot, consts.SysClassNet, ifaceName, "address")
-	data, err := os.ReadFile(macFilePath)
+	addressFilePath := filepath.Join(vars.FilesystemRoot, consts.SysClassNet, ifaceName, "address")
+	data, err := os.ReadFile(addressFilePath)
 	if err != nil {
-		log.Log.Error(err, "GetNetDevMac(): fail to read Mac file", "path", macFilePath)
+		log.Log.Error(err, "readInterfaceAddress(): fail to read address file", "path", addressFilePath)
 		return ""
 	}
 
 	return strings.TrimSpace(string(data))
 }
 
+func (n *network) GetNetDevNodeGUID(pciAddr string) string {
+	log.Log.V(2).Info("GetNetDevNodeGUID(): get node GUID", "pciAddr", pciAddr)
+	if len(pciAddr) > 0 {
+		rdmaDevicesPath := filepath.Join(vars.FilesystemRoot, consts.SysBusPciDevices, pciAddr, "infiniband")
+		rdmaDevices, err := os.ReadDir(rdmaDevicesPath)
+		if err != nil {
+			log.Log.Error(err, "GetNetDevNodeGUID(): failed to read RDMA related directory", "pciAddr", pciAddr)
+			return ""
+		}
+
+		if len(rdmaDevices) != 1 {
+			log.Log.Error(err, "GetNetDevNodeGUID(): expected just one RDMA device", "pciAddr", pciAddr, "numOfDevices", len(rdmaDevices))
+			return ""
+		}
```

Comment:
Is this assumption right?

Reply:
@adrianchiris I am not an expert in RDMA